### PR TITLE
🎉 New destination: Streamr

### PR DESCRIPTION
**Pull request by thanhlmm**
## What
Add Streamr destination connector

## How
Add Streamr destination using `faros-airbyte-cdk`

## Recommended reading order
1. `Readme.md`
2. `resources/spec.json`

## 🚨 User Impact 🚨
No

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
@thanhlmm thanks for the contribution

**Comment by thanhlmm** [comment]
@harshithmullapudi 

You're welcome. Please help me review and point out any remaining work so I can make an update.
Thanks

**Comment by marcosmarxm** [comment]
Hello @thanhlmm this is the first time for a contribution adding a connector using the TypeScript Faros CDK. I'm talking to the team to see the best way to handle, and soon as I got any update I'll let you know.

**Comment by thanhlmm** [comment]
Hi @marcosmarxm
No worries, take your time!

**Comment by harshithmullapudi** [comment]
Hey @thanhlmm I will review this PR by EOW. Sorry for the delay

File: airbyte-integrations/connectors/destination-streamr/destinations/streamr-destination/airbyte_secret/config.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "privateKey": "9c9853f5ad682ae1552cf9bb205da37b704bc6d6105a94e190c5727d25e3e6cc",
+  "streamId": "0x0d0102474519cd2fc1b3e3f962a87e39cbcbead2/test-streamr"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-streamr/destinations/streamr-destination/airbyte_secret/config.json:3]: secrets must be placed inside the `secrets/` folder. @thanhlmm can you remove this?

Comment (thanhlmm) [reply airbyte-integrations/connectors/destination-streamr/destinations/streamr-destination/airbyte_secret/config.json]: Solved. Thanks!

**Comment by thanhlmm** [comment]
Hi @harshithmullapudi, any update on this?

**Comment by fonty1** [comment]
@harshithmullapudi - it would be great to have this merged soon so we can give this integration some attention on our socials :)

**Comment by harshithmullapudi** [comment]
Hey, @fonty1 sorry for the delay. We are just trying to figure out the best way to merge this in our code base. As this will be our first cross-language connector. Will talk to the team and see what we can do.

**Comment by fonty1** [comment]
> Hey, @fonty1 sorry for the delay. We are just trying to figure out the best way to merge this in our code base. As this will be our first cross-language connector. Will talk to the team and see what we can do.

Ah, interesting!

**Comment by harshithmullapudi** [comment]
Also happy to discuss if you have suggestions for example

1. Ideally we don't want all these config files (eslint.js, prettier) for every source. It would be better if we can common files across all sources
2. Also to figure out how to run tests. Before that what is the test package we want to use and how can we integrate this with our source approval flow.

Otherwise I had a basic look at the PR and looked promising.

**Comment by thanhlmm** [comment]
> Also happy to discuss if you have suggestions for example
> 
> 1. Ideally we don't want all these config files (eslint.js, prettier) for every source. It would be better if we can common files across all sources
> 2. Also to figure out how to run tests. Before that what is the test package we want to use and how can we integrate this with our source approval flow.
> 
> Otherwise I had a basic look at the PR and looked promising.

1. I think we can do it with some effort. We need to move these config files to the upper level and have a config file to track only Javascript/Typescript integration only (We don't want eslint or prettier mess others integration)
2. To be honest, I don't see much difference in testing libraries in Javascript. So I think `jest` might be good, it is top library in Javascript world.

**Comment by marcosmarxm** [comment]
Hello @thanhlmm, sorry for the long delay to return to you. I talked to Sherif about adding connectors using externals CDK`s. At the moment looks is better to add your contribution to the TypeScript CDK repo or you keep in your repo. In TypeScript CDK repo will be much more easy to create Action and Workflows to apply tests.

You can publish your connector in Airbyte Connector list. See example from Jenkins PR from Faros.ai https://github.com/airbytehq/airbyte/pull/7837. 

Let me know what do you think about it.

**Comment by thanhlmm** [comment]
Hey @marcosmarxm do you mean that I should keep the source code in my own repo for now, and wait #7837 to be merged?

**Comment by harshithmullapudi** [comment]
Hey @thanhlmm his point is to
1. You can have this as a source in this repo https://github.com/faros-ai/airbyte-connectors or have it in your repo only and leave it for now until we figure out the right way to do this.
2. You can in the meantime use https://github.com/airbytehq/airbyte/pull/7837 to publish the connector.

**Comment by thanhlmm** [comment]
@harshithmullapudi Thanks

Then I will keep the source in my repo and also add it to connector catalog

**Pull request closed without merging**